core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -624,9 +630,13 @@ private void contextProcessedSuccessfully(SearchContext context) {
     }
 
     private void cleanContext(SearchContext context) {
-        assert context == SearchContext.current();
-        context.clearReleasables(Lifetime.PHASE);
-        SearchContext.removeCurrent();
+        try {
+            assert context == SearchContext.current();
+            context.clearReleasables(Lifetime.PHASE);
+            SearchContext.removeCurrent();
+        } finally {
+            context.decRef();
```

Comment:
Actually I think the RC would most of times be 1 after this call to decRef(). It would only be 0 if the context was forcefully freed (because of the reaper or a clear scrall API call) while the request was being executed. So the best we could assert is that the RC is either 0 or 1, but I'm not sure this buys much?